### PR TITLE
Removed obsolete code in fpo.py

### PR DESCRIPTION
**Pull request by claudep**
This code was deprecated since 2009 (ba8e61b93ef0).

### CODE REVIEW

File: translate/storage/fpo.py [hunk 1]
```diff
@@ -329,10 +328,6 @@ def hasplural(self):
         source = self.source
         return isinstance(source, multistring) and len(source.strings) > 1
 
-    def parse(self, src):
-        raise DeprecationWarning("Should not be parsing with a unit")
```
Comment (unho) [inline translate/storage/fpo.py]: It is safe to remove `DeprecationWarning` as well since it is only used here and in some other commented line.

**Comment by unho** [comment]
Will merge after addressing raised nit or replying with some reason not to do that.

File: translate/storage/fpo.py (hunk 1 above)
Comment (claudep) [inline translate/storage/fpo.py]: Remove where?

Comment (unho) [inline translate/storage/fpo.py]: In https://github.com/claudep/translate/blob/remove_fpo_depr/translate/storage/fpo.py#L291

Comment (claudep) [inline translate/storage/fpo.py]: Oh, the commented line? Sure, I can.

**Comment by unho** [comment]
If you rebase this PR will be merged.

**Comment by claudep** [comment]
Rebased. I guess the test failures are unrelated to this patch...

**Comment by unho** [comment]
Merged. Thanks a lot!

**Pull request merged**